pymeasure/display/windows.py

```diff
@@ -389,19 +400,24 @@ def open_experiment(self):
                 else:
                     results = Results.load(filename)
                     experiment = self.new_experiment(results)
-                    experiment.curve.update()
+                    for index in range(len(self.widget_list)):
+                        experiment.curve[index].update_data()
                     experiment.browser_item.progressbar.setValue(100.)
                     self.manager.load(experiment)
                     log.info('Opened data file %s' % filename)
 
     def change_color(self, experiment):
+        for index, wdg in enumerate(self.widget_list):
+            if isinstance(wdg, PlotWidget):
+                curve = experiment.curve[index]
+                break
```

Comment:
Now only for a single `PlotWidget` the color is changed; maybe it would be nice to do this for any `PlotWidget` that is present in the `widget_list` (or maybe even nicer if it would be possible to do this for any tab-widget for which it is relevant, though this might be a bit more difficult as there are so far no other tab-widgets for which it is relevant).

Reply:
Ok, I think you are right. I will add a `set_color` method to the `TabWidget` and add the implementation for `PlotWidget`.